next-app/components/ActivePage.js

```diff
@@ -19,13 +20,28 @@ export default function ActivePage ({page, wordData, setActivePage, setWordData}
             );
 
         case "history":
+            let historyData;
+            if (typeof window !== 'undefined' && window.localStorage) {
+                historyData = JSON.parse(localStorage.getItem("history"));
+                if (historyData) {
+                    historyData.forEach((word) => {
+                        if (word.status) { word.status = ""; };
+                        if (inStorage(word.word, "favourites")) { word.status = "active"; };
+                    });
+                };
+            };
```

Comment:
i guess JSON.parse() in this context is ok, because the data is not too big (?)
but there is a storage API for browser extensions:
https://developer.chrome.com/docs/extensions/reference/storage/#property-local
probably preferable to implement this one